chain/store/store.go

```diff
@@ -210,6 +211,24 @@ func (cs *ChainStore) SubscribeHeadChanges(f func(rev, app []*types.TipSet) erro
 	cs.headChangeNotifs = append(cs.headChangeNotifs, f)
 }
 
+func (cs *ChainStore) IsTipSetValidated(ctx context.Context, ts *types.TipSet) (bool, error) {
+	key := tipsetMessageValidationCacheKeyPrefix.Instance(ts.Key().String())
+
+	return cs.ds.Has(key)
+}
+
+func (cs *ChainStore) MarkTipSetAsValidated(ctx context.Context, ts *types.TipSet) error {
+	key := tipsetMessageValidationCacheKeyPrefix.Instance(ts.Key().String())
+
+	err := cs.ds.Put(key, []byte{0})
```

Comment:
nitpick: We prefer to write if errs in one line when there is only a single return value:
```go
if err := cs.ds.Put(...); err != nil {
```